### DISCUSSION TITLE
GO for iMac Mojave

### DISCUSSION THREAD

**Discussion by midimusic** [GrandOrgue/grandorgue#1143] [Help]
I have just reinstalled MacOS Mojave on my 2015 iMac. This model cannot be updated to the latest version of MacOS so I have decided that Mojave 10.14 is the best option. I used to run GO on this system but no longer have a working copy of GO for iMac. The latest version of GO from github will not run on this system and I cannot yet build form source because the required compiler and other resources are very hard to find and generally reluctant to install and work properly.

I have just looked at the SourceForge download for Mac GO and find that the link has been taken over by some other rubbish website. So where can I get an old GO binary for Mac?

csw900

**Comment by willeke1234** [comment]
The SourceForge download link works for me.
https://sourceforge.net/projects/ourorgan/files/GrandOrgue/GrandOrgue-0.3.1.2321-Darwin.dmg/download

**Comment by midimusic** [comment]
Thanks for finding an alternative link - I still cannot find it even though I now know it is there!! However it does download
a .dmg file (which  I have not yet tried). I will come back with a further comment if it does not install and work properly.

This is the most obvious link and is the one which has been pirated https://bintray.com/e9925248/generic/grandorgue
It is on the main files page of the sourceforge website and it would be a good idea to remove it and replace it with your link.

I have spent many hours trying to compile GO from source on my iMac. The instructions for doing this in the Install.md are
quite useful and may work on a much newer Mac but are only a basic outline of what is really needed for older macs.
Problems:
1: It is necessary to install the Xcode command line tools or brew does not work properly (Xcode itself may not be needed).
2: There are problems with brew errors, lots of them, mainly it fails to make the necessary links. It is also necessary to install gcc
and glib, which brew can do but again fails to make the necessary links.
3: Jack needs flac and this is not found and so it does not work. (I would prefer not to have jack anyway).

And all this is before even trying to compile the actual GO code.

csw900

**Comment by willeke1234** [comment]
In my case the big green download button on https://sourceforge.net/projects/ourorgan/ goes to https://sourceforge.net/projects/ourorgan/files/latest/download
The download files are in the Files tab, folder Grandorgue.
Maybe it's the mirror, JFrog shut down Bintray see https://jfrog.com/blog/into-the-sunset-bintray-jcenter-gocenter-and-chartcenter/

I too spent days building Grandorgue on my Mac. I'm on a 2012 Mac mini and macOS Catalina 10.15. Build.md is outdated, see [prepare-osx.sh](https://github.com/GrandOrgue/grandorgue/blob/master/build-scripts/for-osx/prepare-osx.sh). Had to download 2GB of tools and libraries. CMake can generate a Xcode project but there are wrong settings or bugs somewhere and I had to fix a lot of paths. I added the headers to the Xcode project and now I can build and debug GrandOrgue in Xcode. It's slow and fragile but Jump to Definition and a symbolic debugger are very handy.

**Comment by midimusic** [comment]
I now know why I could not find a/the link for a mac download on sourceforge. The text you get to see depends on what operating system you are using. I generally use windows and in windows the main big button link is for a windows version of GO. When I look at the same page with a Mac the mac version of GO appears on the same big button.

This mac version does download and initially worked OK.

However I have now run into a fatal problem - I changed GO's Audio Output from the default (which is a usb d/a converter output because my internal speakers do not work any more) to another, which was given as an option. This option did not work, it blacked out the audio amplitude bars and locked up GO so that it had to be forcibly closed. Now even after deleting the cache, deleting the app and reinstalling the app it still locks up. I have had a look for its configuration files (which are in ($USER)/AppData/Roaming/ on windows but I cannot find them on the Mac. I think the only solution will be to delete these files. Any idea where they are located on a Mac???

I would not have expected GO to compile on Xcode, I often use CodeBlocks which works on the Mac and would probably be OK also for GO. I too can run Catalina but I think Mojave is a better option for an old Mac.

csw900

**Comment by willeke1234** [comment]
The GrandOrgue settings are in ~/Library/Preferences/GrandOrgueConfig

**Comment by midimusic** [comment]
Thanks - I searched Library for "GrandOrgue" and deleted all files and folders which resulted. This cured the problem.
csw900

**Comment by oleg68** [comment]
@midimusic I've tried to build GrandOrgue for OsX 10.14. Unfortunally I don't have any device where I can to test it. Could you try to run [this build](https://github.com/oleg68/GrandOrgue-official/suites/7948014683/artifacts/339318852) on your iMac 10.14? Thank you.

**Comment by midimusic** [comment]
@oleg68 
Unfortunately it instantly crashes before opening any windows. Crash report attached
[GOreport.pdf](https://github.com/GrandOrgue/grandorgue/files/9422686/GOreport.pdf)
.

**Comment by oleg68** [comment]
@midimusic Thank you. 

Unfortunately wxWidgets compiled for 10.15 does not work on 10.14: wxWidgetImpl::CreateContentView references to `_objc_opt_respondsToSelector` that is missing in 10.14.

Because I don't know whery to get wxWidgets 3.2.0 compiled for 10.14 (brew doesn't support it), I **cann't support 10.14**. I'm sorry.

**Comment by midimusic** [comment]
@oleg68 

I know that wxWidgets can be compiled ON 10.14 as I have done it and already use it for other purposes. There is a potential problem that either 32 or 64 bit versions (or both) can be made and you have to be sure to use the correct one. This also applies to other dependencies which may be installed using brew.

If I get some spare time I may have a go at compiling GO from the latest source - (but cannot say when at the moment). If successful the result could perhaps be added to your binary versions on github.

csw900
